plugin/src/main/groovy/gorm/tools/repository/EntityFieldsHandler.groovy

```diff
@@ -0,0 +1,32 @@
+package gorm.tools.repository
+
+import gorm.tools.beans.BeanPathTools
+import grails.core.GrailsApplication
+import org.springframework.beans.factory.annotation.Autowired
+
+trait EntityFieldsHandler<D> {
+
+    @Autowired
+    GrailsApplication grailsApplication
+
+    abstract Class<D> getEntityClass()
+    List<String> getSelectFieldsConfig(){getFieldsConfig('select')}
+    List<String> getListFieldsConfig(){getFieldsConfig('list') ?: getSelectFieldsConfig()}
+    List<String> getShowFieldsConfig(){getFieldsConfig('show') ?: getSelectFieldsConfig()}
+
+
+    List<String> getFieldsConfig(String type) {
+        grailsApplication.config.domainFields?.get(getEntityClass().simpleName)?.get(type) ?: null
```

Comment:
inject or abstract. I don't want to allow this grailsApplication.config stuff anymore. its get to messy and out of control